### PR TITLE
Fix: avoid default node storage access in `write_quad`

### PR DESCRIPTION
**Pull request by liss-h**
Avoid default node storage access in `write_quad`

### CODE REVIEW

File: private/rdf4cpp/writer/WriteQuad.hpp [hunk 1]
```diff
@@ -39,6 +39,11 @@ bool write_pred(Node const pred, writer::BufWriterParts const writer) {
         return false;                       \
     }
 
+[[nodiscard]] inline bool is_default_graph(Node const &graph) noexcept {
+    auto const g = graph.as_iri();
+    return !g.null() && g.is_default_graph();
```
Comment (bigerl) [inline private/rdf4cpp/writer/WriteQuad.hpp]: why check that it is not the null node?

File: src/rdf4cpp/IRI.cpp [hunk 2]
```diff
@@ -140,8 +140,17 @@ bool IRI::is_iri() const noexcept { return true; }
 
 
 IRI IRI::default_graph(storage::DynNodeStoragePtr node_storage) {
-    return IRI::make_unchecked("", node_storage);
+    auto const id = datatypes::registry::reserved_datatype_ids[datatypes::registry::default_graph_iri];
```
Comment (bigerl) [inline src/rdf4cpp/IRI.cpp:143]: Note to myself: this is strange because of https://github.com/rdf4cpp/rdf4cpp/issues/243

**Review by bigerl — APPROVED** [review]
Thx for the PR. Looks good. I have a little remark below but I am sure that is non-blocking

File: private/rdf4cpp/writer/WriteQuad.hpp (hunk 1 above)
Comment (liss-h) [reply private/rdf4cpp/writer/WriteQuad.hpp]: Good point, might as well directly call the function

Comment (bigerl) [reply private/rdf4cpp/writer/WriteQuad.hpp]: patched

**Pull request merged**